src/utils.cpp

```diff
@@ -1459,6 +1459,36 @@ m_time_t m_mktime(struct tm* stm)
     return mktime(stm);
 }
 
+int m_clock_gettime(clockid_t clock_id, timespec *t)
+{
+    if (!t)
```

Comment:
IMHO, this is an internal method. We can assume the caller will always pass a valid pointer